### PR TITLE
[Feature][Issue-2324/Issue-2420] Support selection on yarn queue when create/edit yarn per-job/application-mode application & yarn-session cluster.

### LINKED ISSUE #2420: [Feature][subtask] Support selection on yarn queue when create/edit yarn per-job/application-mode application & yarn-session cluster.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support selection on yarn queue when create/edit yarn per-job/application-mode application & yarn-session cluster.

### Usage Scenario

_No response_

### Related issues

#2324 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by RocMarshal**
## What changes were proposed in this pull request

Issue Number: close #2420 

## Brief change log

Support selection on yarn queue when create/edit yarn per-job/application-mode application & yarn-session cluster.

## Verifying this change

- This change is already covered by existing tests.
- *Manually verified the change by testing locally.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi, @wolfboys Thanks very much for your review.
And I make some change in the commits 2nd--6th.
Would you like to help to TAL in your free time ?
Thanks~!

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java [hunk 1]
```diff
@@ -1724,4 +1731,52 @@ private String getSavePointed(Application appParam) {
...
+  /**
+   * Judge the execution mode whether is the Yarn PerJob or Application mode with not default or
+   * empty queue label.
+   *
+   * @param application application entity.
+   * @return If the executionMode is (Yarn PerJob or application mode) and the queue label is not
+   *     (empty or default), return true, false else.
+   */
+  @VisibleForTesting
+  public boolean isYarnPerJobAppModeNotDefaultQueue(Application application) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java]: We can change `public` to `private` and method rename to `isYarnNotDefaultQueue`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java [hunk 2]
```diff
@@ -356,4 +361,49 @@ public void delete(FlinkCluster cluster) {
...
+  /**
+   * Judge the execution mode whether is the Yarn session mode with not default or empty queue
+   * label.
+   *
+   * @param cluster cluster.
+   * @return If the executionMode is yarn session mode and the queue label is not (empty or
+   *     default), return true, false else.
+   */
+  @VisibleForTesting
+  public boolean isYarnSessionModeAndNotDefaultQueue(FlinkCluster cluster) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java]: I suggest change `public` to `private`, method rename to `isYarnNotDefaultQueue`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java [hunk 3]
```diff
@@ -356,4 +361,49 @@ public void delete(FlinkCluster cluster) {
...
+    return ExecutionMode.isYarnSessionMode(cluster.getExecutionModeEnum())
+        && !yarnQueueService.isEmptyOrDefaultQueue(cluster.getYarnQueue());
+  }
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java:417]: Would it be better to rename `isEmptyOrDefaultQueue` to `isDefaultQueue` ?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/YarnQueueServiceImpl.java [hunk 4]
```diff
@@ -166,6 +168,40 @@ public void deleteYarnQueue(YarnQueue yarnQueue) {
...
+   * Only check the validation of queue-labelExpression when using yarn application or yarn-session
+   * mode or yarn-perjob mode.
+   *
+   * @param executionMode execution mode.
+   * @param queueLabel queueLabel expression.
+   */
+  @Override
+  public void checkQueueLabelFormatIfNeeded(ExecutionMode executionMode, String queueLabel) {
+    if (ExecutionMode.isYarnMode(executionMode)) {
+      ApiAlertException.throwIfFalse(isValid(queueLabel, true), ERR_FORMAT_HINTS);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/YarnQueueServiceImpl.java:181]: I suggest not to throw exception here, just return the result, it's up to the caller to decide whether it needs to throw an exception

File: streampark-common/src/main/java/org/apache/streampark/common/enums/ExecutionMode.java [hunk 5]
```diff
@@ -78,6 +78,10 @@ public static boolean isYarnMode(ExecutionMode mode) {
     return YARN_PER_JOB.equals(mode) || YARN_APPLICATION.equals(mode) || YARN_SESSION.equals(mode);
   }
 
+  public static boolean isYarnPerJobOrAppMode(ExecutionMode mode) {
```
Comment (wolfboys) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/ExecutionMode.java:83]: If there is only one caller, I suggest this method can be removed

File: streampark-console/streampark-console-webapp/src/views/flink/app/hooks/useFlinkRender.tsx [hunk 6]
```diff
@@ -235,18 +238,24 @@ export const renderOptionsItems = (
 export const renderYarnQueue = ({ model, field }: RenderCallbackParams) => {
   return (
     <div>
-      <Input
+      <ApiSelect
         name="yarnQueue"
-        placeholder={t('flink.app.addAppTips.yarnQueuePlaceholder')}
+        placeholder={t('setting.yarnQueue.placeholder.yarnQueueLabelExpression')}
+        api={fetchYarnQueueList}
+        params={{ page: 1, pageSize: 9999, teamId: getUserTeamId() }}
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/hooks/useFlinkRender.tsx]: We don't need the parameter `teamId`

Comment (RocMarshal) [reply streampark-console/streampark-console-webapp/src/views/flink/app/hooks/useFlinkRender.tsx]: done.

File: streampark-common/src/main/java/org/apache/streampark/common/enums/ExecutionMode.java (hunk 5 above)
Comment (RocMarshal) [reply streampark-common/src/main/java/org/apache/streampark/common/enums/ExecutionMode.java:83]: two caller. I'll add a todo tag.
We would inline it back when the perjob mode is dropped.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java (hunk 1 above)
Comment (RocMarshal) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ApplicationServiceImpl.java]: done.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java (hunk 2 above)
Comment (RocMarshal) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java]: Done.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java (hunk 3 above)
Comment (RocMarshal) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/FlinkClusterServiceImpl.java:417]: done.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/YarnQueueServiceImpl.java (hunk 4 above)
Comment (RocMarshal) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/YarnQueueServiceImpl.java:181]: done.

**Review by RocMarshal — COMMENTED** [review]
@wolfboys Thanks a lot for the review ~
I made some change based on your comments.
And remove some corresponding test cases due to (public->private).

File: streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/yarnQueue.ts [hunk 7]
```diff
@@ -27,6 +27,7 @@ export default {
     yarnQueueLabelExpression: "请输入队列，如 {'{queue} 或 {queue}@{lab1,...}'}",
     description: '请输入描述',
   },
+  selectionHint: "用来快速设置 'yarn.application.name' 与 'yarn.application.node-label'。如果没有可用队列请联系管理员在 'Settings' -> 'Yarn Queue' 页面中创建.",
```
Comment (RocMarshal) [inline streampark-console/streampark-console-webapp/src/locales/lang/zh-CN/setting/yarnQueue.ts]:
```suggestion
  selectionHint: "用来快速设置 'yarn.application.name' 与 'yarn.application.node-label'。如果没有可用队列请联系管理员在 'Settings' -> 'Yarn Queue' 页面中创建",
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/YarnQueueService.java [hunk 8]
```diff
@@ -35,4 +36,12 @@ public interface YarnQueueService extends IService<YarnQueue> {
   void updateYarnQueue(YarnQueue yarnQueue);
 
   void deleteYarnQueue(YarnQueue yarnQueue);
+
+  void checkQueueLabelFormatIfNeeded(ExecutionMode executionMode, String queueLabel);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/YarnQueueService.java]: How about `checkQueueLabel`?

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**